File: API_Post_Call_Simple.py
#REQUIRED PACKAGES
import os
from dotenv import load_dotenv
import requests

load_dotenv(dotenv_path="../.env")

#Python Code
import logging

logging.basicConfig(
    filename="error_log.log",
    level=logging.ERROR,
    format="%(asctime)s %(levelname)s: %(message)s",
    datefmt="%Y-%m-%d %H:%M:%S"
)
logger = logging.getLogger(__name__)


#Required Functions
import time

def delay(n):
    time.sleep(n)


#Set Global Variables
api_endpoint_url = 'https://enter_url_here_for_api_endpoint'


#Call Function
async def api_post_call_simple(json_body):
    try:
        #Set Up Configuration
        headers = {
            'Content-Type': 'application/json',
            'Authorization': f'Bearer {os.getenv("API_TOKEN_FOR_POST_CALL")}'
        }

        #Send POST Request
        response = requests.post(api_endpoint_url, json=json_body, headers=headers)
        response.raise_for_status()

        #Process POST Response
        result = response.json()
        print(result)
        return result

    except Exception as err:
        logger.error(f"There was an error in the api_post_call_simple function: {err}")


#api_post_call_simple()

File: Code_Samples/API_Post_Call_Simple.mjs
//REQUIRED PACKAGES
import dotenv from 'dotenv'
dotenv.config({ path: "../.env" });
import https from 'https';
import axios from 'axios';


//JS Code
import logger from "./Error_Logger_Template.mjs";


//Required Functions
function delay(n) {
    return new Promise(function (resolve){
        setTimeout(resolve, n*1000);
    });
};
  

//Set Global Variables
let api_endpoint_url = 'https://enter_url_here_for_api_endpoint';


//Call Function
async function api_post_call_simple(json_body) {
    try { 

    //Set Up Configuration
        let config = {
            headers: { 
                'Content-Type': 'application/json',
                'Authorization': `Bearer ${process.env.API_TOKEN_FOR_POST_CALL}`
            }
        };
    
    //Send POST Request
        const response = await axios.post(api_endpoint_url, json_body, config);

    //Process POST Response
        const result = response.data;
        console.log(result);
        return result;

    } catch(err) {
        logger.error(`There was an error in the API_Post_Call_Simple.mjs: ` + (err.stack || err));
    };
};


//api_post_call_simple();
export default api_post_call_simple;